src/BackgroundTask/RecurringBackgroundTask.php

```diff
@@ -0,0 +1,55 @@
+<?php
+/**
+ * Abstract class RecurringBackgroundTask.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP\BackgroundTask;
+
+/**
+ * Abstract base class for using cron to execute a background task that runs on a schedule.
+ *
+ * @package AmpProject\AmpWP
+ * @since 2.1
+ * @internal
+ */
+abstract class RecurringBackgroundTask extends CronBasedBackgroundTask {
+
+	/**
+	 * Register the service with the system.
+	 */
+	public function register() {
+		parent::register();
+
+		add_action( 'admin_init', [ $this, 'schedule_event' ] );
+		add_action( $this->get_event_name(), [ $this, 'process' ] );
+	}
+
+	/**
+	 * Schedule the event.
+	 *
+	 * @param mixed[] ...$args Arguments passed to the function from the action hook.
+	 */
+	public function schedule_event( ...$args ) {
+		if ( ! is_user_logged_in() ) {
+			return;
+		}
+
+		$event_name = $this->get_event_name();
+		$timestamp  = wp_next_scheduled( $event_name );
```

Comment:
I found that a `amp_validate_urls` event was getting registered with every request, and the reason is that `wp_next_scheduled( $event_name )` was never returning a timestamp. This is due to the inclusion of `$args` in 1419ef4e3ccb4654d6e6f70e8ddfa19c8e8f68da which also needed to be passed to `wp_next_scheduled()`:

```suggestion
		$timestamp  = wp_next_scheduled( $event_name, $args );
```